crates/key-server/src/server.rs

```diff
@@ -255,7 +257,18 @@ impl Server {
             .await
             .map_err(|e| {
                 warn!("Dry run execution failed ({:?}) (req_id: {:?})", e, req_id);
-                InternalError::Failure
+                // A dry run will fail if called with a newly created object parameter that the FN has not yet seen.
+                // In that case, the user gets a 503 status code (service unavailable).
+                match e {
+                    Error::RpcError(ClientError::Call(e)) => match e.code() {
+                            INVALID_PARAMS_CODE => {
+                                warn!("Invalid parameter: This could be because the FN has not yet seen the object.");
```

Comment:
```suggestion
                                debug!("Invalid parameter: This could be because the FN has not yet seen the object.");
```
(since it's a user error)